prog/probe/main.go

```diff
@@ -101,12 +101,19 @@ func main() {
 	}
 	log.Printf("publishing to: %s", strings.Join(targets, ", "))
 
-	clients := xfer.NewMultiAppClient(xfer.ProbeConfig{
+	probeConfig := xfer.ProbeConfig{
 		Token:    *token,
 		ProbeID:  probeID,
 		Insecure: *insecure,
-	}, xfer.ControlHandlerFunc(controls.HandleControlRequest), xfer.NewAppClient)
+	}
+	clients := xfer.NewMultiAppClient(func(hostname, endpoint string) (xfer.AppClient, error) {
+		return xfer.NewAppClient(
+			probeConfig, hostname, endpoint,
+			xfer.ControlHandlerFunc(controls.HandleControlRequest),
+		)
+	})
 	defer clients.Stop()
+	controls.Client = clients
```

Comment:
Agreed; but plumbing through a PipeFactory into the docker registry seems more naff.  Any other suggestions?